### ISSUE TITLE
info command isn't downloading all deps

### ISSUE THREAD

**Issue by timeyyy** [thheller/shadow-cljs#1216]
i install all deos using `clojure -P`, when i run shadow with info command, i see the info output (no download occurs). when i go to start a shadow server or shadow release it starts downloading dependencies.

` thheller/shadow-cljs        {:mvn/version "2.28.20"}`

**Comment by thheller** [comment]
The `info` command only downloaded dependencies when specified in `shadow-cljs.edn`. For `deps.edn` you can use the regular `clojure` command instead, forgot what the option name was to only download, but `clojure -Spath` would work.

**Comment by thheller** [comment]
Do you actually have shadow-cljs configured to use `deps.edn`? Otherwise it'll try to get the dependencies listed in `shadow-cljs.edn`?

**Comment by timeyyy** [comment]
Yes i have the :deps alias set with a few aliases, and nothing under :dependencies in shadow-cljs file

**Comment by timeyyy** [comment]
Ok i found something out... I'm scripting with babashka, and when i use process to run the clj command line, it also downlaods these same packages...

clj will not cause the download otherwise. 

just to clarify, i am getting the download behavior when using shadow-cljs from my normal environment (not with babashka or as a subprocess)

**Comment by thheller** [comment]
I'm sorry. I have no clue what problem you are describing here anymore.

What exactly is the issue? What does babashka suddenly have to do with any of this?

**Comment by timeyyy** [comment]
Let me rephrase to better connect what I'm seeing:

The core behavior I'm observing is that Google Closure dependencies are being downloaded in two scenarios:
1. When running shadow-cljs (configured to use deps.edn, no dependencies in shadow-cljs.edn)
2. When executing `clj` commands as subprocesses through Babashka

This happens even after running `clojure -P` to prepare dependencies. The fact that both scenarios trigger similar downloads might suggest this is related to a dependency that shadow-clls has on clojure, rather than being specific to shadow-cljs.

The core issue remains: Why is shadow-cljs downloading these dependencies when they should already be prepared by `clojure -P`? 

Anyway i have at least figured out it's something not specific to shadow-cljs so i'll look elsewhere for the fix...

**Comment by thheller** [comment]
Can you be more specific please. What dependencies exactly get downloaded? Why are you mentioning Google Closure specifically?

I still don't understand what the actual issue is? You are expecting no dependencies to be download? When using `deps.edn` then the `clojure` command is solely responsible for downloading all dependencies.

**Comment by timeyyy** [comment]
> What dependencies exactly get downloaded? 
In my second message in this thread, i pasted a subset of them in

> Why are you mentioning Google Closure specifically
Just picked it out because it stuck in my mind, mentioned as an example

> You are expecting no dependencies to be download?
Yes exactly

**Note** just tried this on a hello world setup and couldn't reproduce

**Comment by thheller** [comment]
Are you maybe using aliases but running a blank `clojure -P`? Meaning you have `:deps {:aliases [:foo]}` in your `shadow-cljs.edn` and shadow-cljs being in that alias? If so you'd need to specify those aliases when prepping.

**Comment by thheller** [comment]
Only other guess I have is `:deps` being placed incorrectly in `shadow-cljs.edn` and thus the `shadow-cljs` command not actually trying to use the `clojure` command. That may end up with a different `shadow-cljs` version from `package.json` instead, thus leading to the extra downloads.

**Comment by thheller** [comment]
Closing as this doesn't seem to be shadow-cljs related. When using `:deps` key in `shadow-cljs.edn` then the `shadow-cljs` command does not perform any dependency downloading whatsoever. It is all entirely up to the `clojure` command with whatever aliases were configured.
